src/frontend/src/utils/condition.rs

```diff
@@ -849,6 +849,7 @@ impl Condition {
             .into_iter()
             .map(push_down_not)
             .map(fold_boolean_constant)
+            .map(column_self_eq_eliminate)
```

Comment:
the reason I put this optimization here is because it's hard to catch new variants of predicate(s) after each `predicate_pushdown` in `logical_optimization`, and the predicate in `ch_benchmark` can't be optimized though I've tried to insert as many `LOGICAL_FILTER_EXPRESSION_SIMPLIFY` as possible.

thus it's better to solve this problem from the root - from my perspective.